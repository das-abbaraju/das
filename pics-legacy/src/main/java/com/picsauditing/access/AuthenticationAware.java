package com.picsauditing.access;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface AuthenticationAware {
}
