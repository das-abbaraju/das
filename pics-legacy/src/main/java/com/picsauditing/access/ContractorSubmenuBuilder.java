package com.picsauditing.access;

import com.picsauditing.actions.TranslationActionSupport;
import com.picsauditing.actions.contractors.AuditMenuBuilder;
import com.picsauditing.jpa.entities.ContractorAccount;
import com.picsauditing.report.RecordNotFoundException;
import com.picsauditing.service.i18n.TranslationServiceFactory;
import org.perf4j.StopWatch;
import org.perf4j.slf4j.Slf4JStopWatch;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;
import java.util.Map;

public final class ContractorSubmenuBuilder {

	private static final Logger logger = LoggerFactory.getLogger(ContractorSubmenuBuilder.class);
	public static final String NOTES = "global.Notes";
	public static final String BILLING_DETAIL = "BillingDetail.title";
	public static final String COMPANY = "global.Company";
	public static final String TRADES = "global.Trades";
	public static final String DOCUGUARD = "menu.DocuGUARD";
	public static final String AUDITGUARD = "menu.AuditGUARD";
	public static final String MANUAL_AUDIT = "AuditType.2.name";
	public static final String PQF = "AuditType.1.name";
	public static final String IMPLEMENTATION_AUDIT = "AuditType.3.name";
	public static final String INSUREGUARD = "menu.InsureGUARD";
	public static final String MANAGE_CERTIFICATES = "ContractorActionSupport.ManageCertificates";
	public static final String AUTOMOBILE_LIABILITY = "AuditType.15.name";
	public static final String EXCESS_UMBRELLA_LIABILITY = "AuditType.16.name";
	public static final String GENERAL_LIABILITY = "AuditType.13.name";
	public static final String WORKER_COMP = "AuditType.14.name";
	public static final String EMPLOYEEGUARD = "menu.EmployeeGUARD";
	public static final String CLIENT_REVIEW = "menu.ClientReview";

	public ContractorSubmenuBuilder() {
	}

	public MenuComponent buildMenubar(ContractorAccount contractorAccount, Permissions permissions,
	                                  boolean includeSupportMenu) throws RecordNotFoundException {
		Logger profiler = LoggerFactory.getLogger("org.perf4j.DebugTimingLogger");
		StopWatch stopwatch = new Slf4JStopWatch(profiler);
		stopwatch.start("ContractorSubmenuBuilder.buildMenubar");

		AuditMenuBuilder auditMenuBuilder = new AuditMenuBuilder(contractorAccount, permissions);
		Map<AuditMenuBuilder.Service, List<MenuComponent>> serviceMenus = auditMenuBuilder.buildAuditMenu();

		MenuComponent menubar = new MenuComponent();
		addChildrenToMenubar(menubar, getText(DOCUGUARD), serviceMenus.get(AuditMenuBuilder.Service.DOCUGUARD));
		addChildrenToMenubar(menubar, getText(AUDITGUARD), serviceMenus.get(AuditMenuBuilder.Service.AUDITGUARD));
		addChildrenToMenubar(menubar, getText(INSUREGUARD), serviceMenus.get(AuditMenuBuilder.Service.INSUREGUARD));
		addChildrenToMenubar(menubar, getText(EMPLOYEEGUARD), serviceMenus.get(AuditMenuBuilder.Service.EMPLOYEEGUARD));
		addChildrenToMenubar(menubar, getText(CLIENT_REVIEW), serviceMenus.get(AuditMenuBuilder.Service.CLIENT_REVIEWS));
		addSupportMenu(menubar, permissions, contractorAccount, includeSupportMenu);

		stopwatch.stop("ContractorSubmenuBuilder.buildMenubar");
		return menubar;
	}

	private MenuComponent addChildrenToMenubar(MenuComponent menubar, String title, List<MenuComponent> menuItems) {
		if (menuItems != null && !menuItems.isEmpty()) {
			MenuComponent menuEntry = menubar.addChild(title);
			for (MenuComponent item : menuItems) {
				menuEntry.addChild(item);
			}
            MenuBuilder.removeMenuIfEmpty(menubar,menuEntry);
		}

		return menubar;
	}

	// todo: Revisit. For now, this is basically the same as the Support item on
	// main menu. We'll see if the two menus
	// end up diverging or not.
	protected MenuComponent addSupportMenu(MenuComponent menubar, Permissions permissions,
	                                       ContractorAccount contractorAccount, boolean includeSupportMenu) {
		if (includeSupportMenu) {
			MenuComponent supportMenu = menubar.addChild(getText("menu.Support"));

			supportMenu.addChild(getText("Header.HelpCenter"), "HelpCenter.action", "help_center");
			supportMenu.addChild(getText("Registration.Error.LiveChat"), "#", "live_chat");
			supportMenu.addChild(getText("global.ContactPICS"), "Contact.action", "contact_action");
			supportMenu.addChild(getText("global.AboutPICS"), "About.action", "about_pics");

			MenuComponent referenceMenu = supportMenu.addChild("Reference");
			if (permissions.hasPermission(OpPerms.ManageTrades)) {
				referenceMenu.addChild(getText("TradeTaxonomy.title"), "TradeTaxonomy.action", "TradeTaxonomy");
			}

			referenceMenu.addChild("Navigation Menu", "Reference!navigationMenu.action", "navigation_menu");
			referenceMenu.addChild("Dynamic Reports", "Reference!dynamicReport.action", "dynamic_report");
			referenceMenu.addChild("Reports Manager", "Reference!reportsManager.action", "reports_manager");
			/*
			 * referenceMenu.addChild("Navigation Restructure",
			 * "Reference!navigationRestructure.action",
			 * "navigation_restructure");
			 */
		}

		return menubar;
	}

	private static String getText(String key) {
		return TranslationServiceFactory.getTranslationService().getText(key, TranslationActionSupport.getLocaleStatic());
	}

}
