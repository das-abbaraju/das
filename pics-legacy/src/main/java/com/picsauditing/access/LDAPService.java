package com.picsauditing.access;

import com.picsauditing.featuretoggle.Features;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.ldap.authentication.ad.ActiveDirectoryLdapAuthenticationProvider;

import javax.security.auth.login.FailedLoginException;


public class LDAPService {

    public static final String PICS_CORP = "@PICS.CORP";

    private static final Logger logger = LoggerFactory.getLogger(LDAPService.class);

    public static final String PICSAD = "picsad";

    @Autowired
    private ActiveDirectoryLdapAuthenticationProvider ldapActiveDirectoryAuthProvider;

    public boolean doLDAPLoginAuthentication(String identityProvider, String username, String password) throws FailedLoginException {
        if (Features.USE_LDAP_AUTHENTICATION.isActive()) {
            switch (identityProvider) {
                case PICSAD:
                    return doPICSLdapAuthentication(username, password);
            }
        }
        return false;
    }

    private String getPICSLdapUser(String username) {
        StringBuilder ldapUser = new StringBuilder();
        ldapUser.append(username);
        ldapUser.append(PICS_CORP);
        return ldapUser.toString();
    }

    private boolean doPICSLdapAuthentication(String username, String password) {
        ldapActiveDirectoryAuthProvider.setConvertSubErrorCodesToExceptions(true);
        String ldapUser = getPICSLdapUser(username);
        try {
            UsernamePasswordAuthenticationToken authentication = new UsernamePasswordAuthenticationToken(ldapUser, password);
            Authentication result = ldapActiveDirectoryAuthProvider.authenticate(authentication);
            return result != null ? result.isAuthenticated() : false;
        } catch (AuthenticationException ace) {
            logger.error("Bad LDAP Credentials for user: " + username);
        }
        return false;
    }
}
