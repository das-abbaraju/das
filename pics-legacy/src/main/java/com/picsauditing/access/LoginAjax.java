package com.picsauditing.access;

import org.json.simple.JSONObject;

import com.picsauditing.actions.PicsActionSupport;

@SuppressWarnings("serial")
public class LoginAjax extends PicsActionSupport {

	private String callback;

	@SuppressWarnings("unchecked")
	@Anonymous
	@Override
	public String execute() throws Exception {
		loadPermissions(false);
		json = new JSONObject();
		json.put("loggedin", permissions.isLoggedIn());
		json.put("user", permissions.getName());
		return JSONP;
	}

	public String getCallback() {
		return callback;
	}

	public void setCallback(String callback) {
		this.callback = callback;
	}
}
