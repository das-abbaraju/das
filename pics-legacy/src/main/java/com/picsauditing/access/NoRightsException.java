package com.picsauditing.access;

import com.picsauditing.actions.TranslationActionSupport;
import com.picsauditing.service.i18n.TranslationServiceFactory;

public class NoRightsException extends Exception {
	private static final long serialVersionUID = 263711643706157627L;

	public NoRightsException(OpPerms opPerm, OpType oType) {
		super(TranslationServiceFactory.getTranslationService().getText("Exception.NoRights",
				TranslationActionSupport.getLocaleStatic(), oType, opPerm));
	}

	public NoRightsException(String groupName) {
		super(TranslationServiceFactory.getTranslationService().getText("Exception.NoRights.GroupName",
				TranslationActionSupport.getLocaleStatic(), groupName));
	}

}
