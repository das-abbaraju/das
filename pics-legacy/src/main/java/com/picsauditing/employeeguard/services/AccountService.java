package com.picsauditing.employeeguard.services;

import com.picsauditing.PICS.Utilities;
import com.picsauditing.dao.AccountDAO;
import com.picsauditing.dao.ContractorAccountDAO;
import com.picsauditing.dao.OperatorAccountDAO;
import com.picsauditing.employeeguard.daos.AccountEmployeeGuardDAO;
import com.picsauditing.employeeguard.entities.Employee;
import com.picsauditing.employeeguard.entities.Profile;
import com.picsauditing.employeeguard.services.external.BillingService;
import com.picsauditing.employeeguard.services.models.AccountModel;
import com.picsauditing.employeeguard.services.models.AccountType;
import com.picsauditing.employeeguard.util.Extractor;
import com.picsauditing.employeeguard.util.ExtractorUtil;
import com.picsauditing.jpa.entities.Account;
import com.picsauditing.jpa.entities.ContractorAccount;
import com.picsauditing.jpa.entities.ContractorOperator;
import com.picsauditing.jpa.entities.OperatorAccount;
import com.picsauditing.provisioning.ProductSubscriptionService;
import org.apache.commons.collections.CollectionUtils;
import org.apache.commons.lang3.ArrayUtils;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.*;

/**
 * This is a stand-in for a "remote" service call
 */
public class AccountService {

	@Autowired
	private AccountDAO accountDAO;
	@Autowired
	private AccountEmployeeGuardDAO accountEmployeeGuardDAO;
	@Autowired
	private ContractorAccountDAO contractorDAO;
	@Autowired
	private OperatorAccountDAO operatorDAO;
	@Autowired
	private BillingService billingService;
	@Autowired
	private ProductSubscriptionService productSubscriptionService;

	public AccountModel getAccountById(int accountId) {
		Account account = accountDAO.find(accountId);
		return mapAccountToAccountModel(account);
	}

	public List<AccountModel> getAccountsByIds(Collection<Integer> accountIds) {
		List<Account> accounts = accountDAO.findByIds(accountIds);
		return mapAccountsToAccountModels(accounts);
	}

	public List<AccountModel> getTopmostCorporateAccounts(final int accountId) {
		if (accountId <= 0) {
			throw new IllegalArgumentException("Invalid account ID: " + accountId);
		}

		List<OperatorAccount> employeeGUARDCorporates = getEmployeeGUARDCorporates(Arrays.asList(accountId));
		// We don't have a need to modify accounts, so we'll map these corporate accounts to AccountModels
		return mapAccountsToAccountModels(employeeGUARDCorporates);
	}

	public Map<Integer, AccountModel> getIdToAccountModelMap(final Collection<Integer> accountIds) {
		return Utilities.convertToMap(getAccountsByIds(accountIds), new Utilities.MapConvertable<Integer, AccountModel>() {

			@Override
			public Integer getKey(AccountModel accountModel) {
				return accountModel.getId();
			}
		});
	}

	public List<Integer> getTopmostCorporateAccountIds(final Collection<Integer> accountIds) {
		if (CollectionUtils.isEmpty(accountIds)) {
			return Collections.emptyList();
		}

		return extractIdFromAccountModel(getEmployeeGUARDCorporates(accountIds));
	}

	public List<Integer> getTopmostCorporateAccountIds(final int accountId) {
		if (accountId <= 0) {
			throw new IllegalArgumentException("Invalid account ID: " + accountId);
		}

		return extractIdFromAccountModel(getEmployeeGUARDCorporates(Arrays.asList(accountId)));
	}

	private List<OperatorAccount> getEmployeeGUARDCorporates(Collection<Integer> accountIds) {
		List<OperatorAccount> operators = operatorDAO.findOperators(new ArrayList<>(accountIds));

		ArrayList<Integer> visited = new ArrayList<>();

		Set<OperatorAccount> corporates = new HashSet<>();
		for (OperatorAccount operator : operators) {
			List<OperatorAccount> topmostCorporates = getTopmostCorporates(operator, visited);
			corporates.addAll(billingService.filterEmployeeGUARDAccounts(topmostCorporates));
		}

		return new ArrayList<>(corporates);
	}

	private List<OperatorAccount> getTopmostCorporates(OperatorAccount operator, List<Integer> visited) {
		List<OperatorAccount> corporates = new ArrayList<>();
		if (onlyHasPicsConsortiumOrNoParents(operator)) {
			visited.add(operator.getId());
			corporates.add(operator);
		} else {
			for (OperatorAccount parent : operator.getParentOperators()) {
				if (!parent.isInPicsConsortium() && !visited.contains(parent.getId())) {
					visited.add(parent.getId());
					corporates.addAll(getTopmostCorporates(parent, visited));
				}
			}
		}

		return corporates;
	}

	private boolean onlyHasPicsConsortiumOrNoParents(OperatorAccount operator) {
		if (operator.getParentOperators().size() == 0) {
			return true;
		}

		for (OperatorAccount parent : operator.getParentOperators()) {
			if (!parent.isInPicsConsortium()) {
				return false;
			}
		}

		return true;
	}

	public List<AccountModel> getChildOperators(final int accountId) {
		return getChildOperators(Arrays.asList(accountId));
	}

	public List<AccountModel> getChildOperators(final List<Integer> accountIds) {
		List<OperatorAccount> corporates = operatorDAO.findOperators(accountIds);
		if (CollectionUtils.isEmpty(corporates)) {
			return Collections.emptyList();
		}

		Set<OperatorAccount> childAccounts = new HashSet<>();
		for (OperatorAccount corporate : corporates) {
			List<OperatorAccount> childOperators = new ArrayList<>(corporate.getChildOperators());
			childAccounts.addAll(billingService.filterEmployeeGUARDAccounts(childOperators));
		}

		return mapAccountsToAccountModels(new ArrayList<>(childAccounts));
	}

	public List<Integer> getChildOperatorIds(final int accountId) {
		return extractIdFromAccountModel(getChildOperators(accountId).toArray(new AccountModel[0]));
	}

	public List<Integer> getChildOperatorIds(final List<Integer> accountIds) {
		return extractIdFromAccountModel(getChildOperators(accountIds).toArray(new AccountModel[0]));
	}

	public AccountType getAccountTypeByUserID(int userID) {
		return getAccountByUserID(userID).getAccountType();
	}

	public AccountModel getAccountByUserID(int userID) {
		return getAccountById(accountDAO.findByUserID(userID));
	}

	public Map<Integer, AccountModel> getContractorMapForSite(final int siteId) {
		return Utilities.convertToMap(getContractors(siteId), new Utilities.MapConvertable<Integer, AccountModel>() {
			@Override
			public Integer getKey(AccountModel accountModel) {
				return accountModel.getId();
			}
		});
	}

	/**
	 * Retrieve the list of Contractor IDs that work under this Operator/Corporate hierarchy and have EmployeeGUARD.
	 *
	 * @param siteId The ID of the Operator Facility/Site or the ID of the Corporation
	 * @return
	 */
	public List<Integer> getContractorIds(final int siteId) {
		return extractIdFromAccountModel(getContractors(siteId).toArray(new AccountModel[0]));
	}

	public List<Integer> getContractorIds(final Collection<Integer> siteIds) {
		if (CollectionUtils.isEmpty(siteIds)) {
			return Collections.emptyList();
		}

		return contractorDAO.findAllContractorIdsForOperatorIds(siteIds);
	}

	/**
	 * Retrieve the list of Contractors that work under this Operator/Corporate hierarchy and have EmployeeGUARD.
	 *
	 * @param siteId The ID of the Operator Facility/Site or the ID of the Corporation
	 * @return
	 */
	public List<AccountModel> getContractors(final int siteId) {
		OperatorAccount operator = operatorDAO.find(siteId);
		if (operator == null) {
			return Collections.emptyList();
		}

		List<ContractorAccount> contractors = new ArrayList<>();
		if (!operator.isCorporate()) {
			contractors = contractorDAO.findAllContractorsForOperator(operator);
		} else {
			for (OperatorAccount site : operator.getChildOperators()) {
				addContractorsFromOperator(site, contractors);
			}
		}

		List<ContractorAccount> accounts = billingService.filterEmployeeGUARDAccounts(contractors);

		return mapAccountsToAccountModels(accounts);
	}

	private void addContractorsFromOperator(OperatorAccount operator, List<ContractorAccount> contractors) {
		for (ContractorOperator contractorOperator : operator.getContractorOperators()) {
			contractors.add(contractorOperator.getContractorAccount());
		}
	}

	private <E extends Account> List<AccountModel> mapAccountsToAccountModels(List<E> accounts) {
		if (CollectionUtils.isEmpty(accounts)) {
			return Collections.emptyList();
		}

		List<AccountModel> accountModels = new ArrayList<>(accounts.size());
		for (Account account : accounts) {
			accountModels.add(mapAccountToAccountModel(account));
		}

		return accountModels;
	}

	private AccountModel mapAccountToAccountModel(Account account) {
		return new AccountModel.Builder().accountType(getAccountTypeForAccount(account)).id(account.getId())
				.name(account.getName()).build();
	}

	private AccountType getAccountTypeForAccount(Account account) {
		switch (account.getType()) {
			case "Admin":
				return AccountType.ADMIN_ACCOUNT;

			case "Assessment":
				return AccountType.ASSESSMENT;

			case "Contractor":
				return AccountType.CONTRACTOR;

			case "Corporate":
				return AccountType.CORPORATE;

			case "Operator":
				return AccountType.OPERATOR;

			default:
				throw new IllegalArgumentException("Invalid account type " + account.getType());
		}
	}

	public List<Integer> extractIdFromAccountModel(final AccountModel... accountModels) {
		if (ArrayUtils.isEmpty(accountModels)) {
			return Collections.emptyList();
		}

		List<Integer> ids = new ArrayList<>();
		for (AccountModel accountModel : accountModels) {
			ids.add(accountModel.getId());
		}

		return ids;
	}

	private List<Integer> extractIdFromAccountModel(List<? extends Account> accounts) {
		if (CollectionUtils.isEmpty(accounts)) {
			return Collections.emptyList();
		}

		List<Integer> ids = new ArrayList<>();
		for (Account account : accounts) {
			ids.add(account.getId());
		}

		return ids;
	}

	public boolean doesContractorStillNeedEmployeeGuard(final int contractorID) {
		return doesContractorStillNeedEmployeeGuard(accountDAO.find(contractorID));
	}

	private boolean doesContractorStillNeedEmployeeGuard(final Account account) {
		return productSubscriptionService.hasEmployeeGUARD(account);
	}

	public Map<Integer, AccountModel> getContractorsForEmployee(final Employee employee) {
		if (employee == null) {
			return Collections.emptyMap();
		}

		if (employee.getProfile() != null) {
			return getContractorsForProfile(employee.getProfile());
		}

		return getIdToAccountModelMap(Arrays.asList(employee.getAccountId()));
	}

	public Map<Integer, AccountModel> getContractorsForProfile(final Profile profile) {
		if (profile == null) {
			return Collections.emptyMap();
		}

		List<Integer> contractorIds = ExtractorUtil.extractList(profile.getEmployees(),
				new Extractor<Employee, Integer>() {
					@Override
					public Integer extract(Employee employee) {
						return employee.getAccountId();
					}
				});

		return getIdToAccountModelMap(contractorIds);
	}

	public Map<Integer, AccountModel> getContractorsForEmployeesMap(final List<Employee> employees) {
		return getIdToAccountModelMap(getAccountIds(employees));
	}

	public List<AccountModel> getContractorsForEmployees(final List<Employee> employees) {
		if (CollectionUtils.isEmpty(employees)) {
			return Collections.emptyList();
		}

		return getAccountsByIds(getAccountIds(employees));
	}

	private List<Integer> getAccountIds(final List<Employee> employees) {
		List<Integer> accountIds = new ArrayList<>();
		for (Employee employee : employees) {
			accountIds.add(employee.getAccountId());
		}

		return accountIds;
	}

	public List<Integer> getOperatorIdsForContractor(final int contractorId) {
		ContractorAccount contractor = contractorDAO.find(contractorId);
		if (contractor == null) {
			return Collections.emptyList();
		}

		return operatorDAO.findAllOperatorsForContractor(contractor);
	}
}
