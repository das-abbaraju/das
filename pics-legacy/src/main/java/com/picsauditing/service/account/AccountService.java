package com.picsauditing.service.account;

import com.picsauditing.jpa.entities.ContractorAccount;
import com.picsauditing.service.account.events.ContractorEventPublisher;
import com.picsauditing.service.account.events.ContractorEventType;
import com.picsauditing.service.account.events.SpringContractorEvent;
import org.springframework.beans.factory.annotation.Autowired;

public class AccountService {

    private final ContractorEventPublisher eventPublisher;

    public AccountService(
            ContractorEventPublisher publisher
    ) {
        this.eventPublisher = publisher;
    }

    public void publishEvent(ContractorAccount account, ContractorEventType type) {
        eventPublisher.publishEvent(account, type);
    }

}
