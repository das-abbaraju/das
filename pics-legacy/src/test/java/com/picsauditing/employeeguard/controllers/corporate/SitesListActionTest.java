package com.picsauditing.employeeguard.controllers.corporate;

import com.picsauditing.PicsActionTest;
import com.picsauditing.access.NoRightsException;
import com.picsauditing.actions.PicsActionSupport;
import com.picsauditing.employeeguard.services.external.AccountService;
import com.picsauditing.employeeguard.services.factory.AccountServiceFactory;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Before;
import org.junit.Test;
import org.mockito.MockitoAnnotations;
import org.powermock.reflect.Whitebox;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.mockito.Mockito.when;

@UseReporter(DiffReporter.class)
public class SitesListActionTest extends PicsActionTest {

	public static final int CONTRACTOR_ID = 234;
	public static final int SITE_ID = 123;

	SitesListAction sitesListAction;

	private AccountService accountService = AccountServiceFactory.getAccountService();

	@Before
	public void setUp() throws Exception {
		MockitoAnnotations.initMocks(this);

		sitesListAction = new SitesListAction();
		super.setUp(sitesListAction);

		when(permissions.getAccountId()).thenReturn(SITE_ID);
		when(permissions.getAccountName()).thenReturn("Site Name");

		Whitebox.setInternalState(sitesListAction, "accountService", accountService);
	}

	@Test(expected = NoRightsException.class)
	public void testList_Site() throws Exception {
		when(permissions.isContractor()).thenReturn(true);

		sitesListAction.list();
	}

	@Test
	public void testList_Corporate() throws Exception {
		when(permissions.isCorporate()).thenReturn(true);

		assertEquals(PicsActionSupport.JSON_STRING, sitesListAction.list());
		assertNotNull(sitesListAction.getJsonString());
		Approvals.verify(sitesListAction.getJsonString());
	}
}
