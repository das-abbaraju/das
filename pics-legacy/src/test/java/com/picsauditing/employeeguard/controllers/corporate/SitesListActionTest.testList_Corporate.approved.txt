[{"id":"1100","name":"Operator"}]